linode/manager
# M3-5118: Linode Create flow filtering for Bare Metal plans

## Description
Bare Metal plans are not available across all regions presently. Like with GPU Linodes, this should be indicated in the Linode Create flow when a user goes to the Bare Metal tab for a region that those plans are not available in:

<img width="913" alt="Screen Shot 2021-08-05 at 12 41 05 PM" src="https://user-images.githubusercontent.com/32860776/128388351-0d817c08-19c2-4059-b6db-aef818a1971f.png">

Also added `doesRegionSupportFeature.ts`, which is more flexible than the separate files that check whether specific individual features are supported by a particular region or not, and eliminates the need for those files.

## How to test
Pointing to dev, begin the Linode Create flow. Select a region other than Newark and you should see the view above; select Newark and you should be able to proceed normally (full creation dependent on back-end considerations).

Confirm that the "Attach VLAN" section has not been adversely impacted by the small refactors that touched it.

**How do I run relevant unit tests?**
`yarn test doesRegionSupportFeature`
